src/Resolvers/Microsoft.DotNet.NativeWrapper/Interop.cs

```diff
@@ -112,19 +152,60 @@ internal static class Unix
                 string exe_dir,
                 hostfxr_get_available_sdks_result_fn result);
 
+            [UnmanagedFunctionPointer(CallingConvention.Cdecl, CharSet = UTF8)]
+            internal delegate void hostfxr_get_dotnet_environment_info_result_fn(
```

Comment:
Why not declare these once and just use CharSet Auto? It feels weird to have to duplicate the declarations if they are essentially the same.

Reply:
Hi all,  I was checking something in this code while chatting with Daniel and happened upon this change. Just for the record, in .NET Core 2.2 and earlier, Auto was unfortunately UTF16 on Unix. I think you can change the others too that I did this way for that reason once upon a time. :)

https://docs.microsoft.com/en-us/dotnet/standard/native-interop/charset

Reply:
Cool feature, by the way! Hope you're all doing well. :)

Reply:
Sounds good, thanks for the background Nick!